exporter/exporterhelper/queued_retry_inmemory.go

```diff
@@ -71,49 +71,136 @@ func (qCfg *QueueSettings) Validate() error {
 	return nil
 }
 
+var (
+	errNoStorageClient        = errors.New("no storage client extension found")
+	errMultipleStorageClients = errors.New("multiple storage extensions found")
+)
+
 type queuedRetrySender struct {
-	fullName        string
-	cfg             QueueSettings
-	consumerSender  requestSender
-	queue           internal.ProducerConsumerQueue
-	retryStopCh     chan struct{}
-	traceAttributes []attribute.KeyValue
-	logger          *zap.Logger
+	id                 config.ComponentID
+	signal             config.DataType
+	cfg                QueueSettings
+	consumerSender     requestSender
+	queue              internal.ProducerConsumerQueue
+	retryStopCh        chan struct{}
+	traceAttributes    []attribute.KeyValue
+	logger             *zap.Logger
+	requeuingEnabled   bool
+	requestUnmarshaler internal.RequestUnmarshaler
 }
 
-func newQueuedRetrySender(id config.ComponentID, _ config.DataType, qCfg QueueSettings, rCfg RetrySettings, _ internal.RequestUnmarshaler, nextSender requestSender, logger *zap.Logger) *queuedRetrySender {
+func (qrs *queuedRetrySender) fullName() string {
+	if qrs.signal == "" {
+		return qrs.id.String()
+	}
+	return fmt.Sprintf("%s-%s", qrs.id.String(), qrs.signal)
+}
```

Comment:
Each signal must have a different name for the persistent queue as otherwise we would be writing multiple signal types to the same underlying storage. In any case, I changed the code a bit so for in-memory queue the name is not depending on the signal type (and stays as it was before)

Reply:
The name is also used for the "queue_size" label, that's why I don't like this change because you are changing the metrics that we emit.

Reply:
So, you need to rename the func to "buildStorageName" and call it only when calling the constructor of the `NewPersistenQueue` the one in the struct `fullName` should be kept the same since we use that in the metrics.